### ISSUE TITLE
L003: Doubled indents are tolerated 

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#3069]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

L003 seems to have an odd behaviour where double indents are tolerated (and sometimes even `fix` ed to).

Consider the following two examples:

```sql
SELECT GREATEST(1, 2,
        3, 4,
    5, 6
)
```

After running `sqlfluff fix`

```sql
SELECT GREATEST(1, 2,
        3, 4,
        5, 6
)
```

Everything gets indented to 2x 4 spaces.

Interestingly, this seems to be specific to 2x. For example tripple (or quadruple) indents get indented correctly

```sql
SELECT GREATEST(1, 2,
            3, 4,
    5, 6
);

SELECT GREATEST(1, 2,
                    3, 4,
    5, 6
);
```

After running `sqlfluff fix`

```sql
SELECT GREATEST(1, 2,
    3, 4,
    5, 6
);

SELECT GREATEST(1, 2,
    3, 4,
    5, 6
);
```

### Expected Behaviour

Doubled indents should not be tolerated (and certainly not suggested as a fix 😉 )

### Observed Behaviour

SQLFluff does not complain about doubled indents.

### How to reproduce

https://online.sqlfluff.com/fluffed?sql=H4sIANP3U2IC_wt29XF1DlFwD3J1DHENDtEw1FEw0uHlUoACYx0FEyjXVEfBjJdL05qXi5crGL8uynTiNQEAsiH7abAAAAA%3D&dialect=ansi

### Dialect

ansi

### Version

0.12 -> webversion

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
Yeah I think this is a limitation with the current reflow rules.

Either this should correct to:
```sql
SELECT GREATEST(
    1, 2,
    3, 4,
    5, 6
);
```

or (the "hanging indent" solution):

```sql
SELECT GREATEST(1, 2,
                3, 4,
                5, 6
);
```

We've had enough issues with hanging indents that I think the _correct_ fix is the first one. The current L003 isn't smart enough to do that. I'll try and address this as part of #3673 .

**Comment by alanmcruickshank** [comment]
This is resolved in #3942. The first example will now be corrected to:
```sql
SELECT GREATEST(
    1, 2,
    3, 4,
    5, 6
)
```
